content/use-cases/issue-certificates.markdown

```diff
@@ -0,0 +1,114 @@
+      )
+      certificate = App::DnsimpleAdapter.issue_letsencrypt_certificate(parsed_request_body['domain'],
+                                                                       certificate_order.certificate_id)
+
+      json(data: {
+             certificate_id: certificate.id,
+             common_name: certificate.common_name,
+             san: certificate.alternate_names,
+             state: certificate.state,
+             auto_renew: certificate.auto_renew
+           })
+    rescue ::Dnsimple::NotFoundError, ::Dnsimple::RequestError => e
+      halt 400, json(error: { message: e.message })
+    end
+~~~
+
+As you can see in our sample application we expose an endpoint `http://localhost:4567/dnsimple/issue_certificate` that takes the following parameters:
+
+* `domain` - The Domain name in your account for which you want to issue a certificate e.g. `pizza.company`
+
+* `name` - Certificate name, where `*` can be used for wildcard certificates.
+
+* `san` - A certificate can be purchased for multiple subdomains. We call them alternate names or [Subject Alternative Name (SAN)](https://support.dnsimple.com/articles/what-is-ssl-san/).
+
+* `auto_renewal` - If a certificate should be auto-renewed by DNSimple.
+
+You can find more about the specific parameters on the certificate resource API page [/v2/certificates](https://developer.dnsimple.com/v2/certificates/#purchaseLetsencryptCertificate).
+
+## Handling Certificate Issue event
+
+As the certificate issue process is asynchronous and can take a while until the certificate is processed by LetsEncrypt, DNSimple sends out an event `certificate.issue` when processing is complete.
+The event contains useful information such as the certificate ID which can be used to retrieve and install it.
+
+[lib/app/handlers/certificate_issue_event_handler.rb](https://github.com/dnsimple/dnsimple-api-examples/blob/main/ruby/use_cases/issue_certificates/lib/app/handlers/certificate_issue_event_handler.rb#L8)
+
+~~~ruby
+      def handle(event)
+        certificate = App::DnsimpleAdapter.download_certificate(
+          event.data['certificate']['domain_id'],
+          event.data['certificate']['id']
+        )
+        private_key = App::DnsimpleAdapter.certificate_private_key(
+          event.data['certificate']['domain_id'],
+          event.data['certificate']['id']
+        ).private_key
+
+        # Now you can carry out your business logic of pushing the certificate to
+        # your infrastructure automation tool and secrets store.
+      end
+~~~
+
+<note>
+Webhooks come with no order guarantee and at least once delivery guarantee, which means that a webhook can be sent multiple times. That is why you can use the `request_identifier` which is present in every event to keep track of which events have already been processed. It is also recommended to only use the underlying resource ID provided in the webhook event data to retrieve the resource, to protect yourself from forged events.
```

Comment:
> at least once delivery guarantee 

This word order is a little confusing. Does this mean that they're guaranteed to be delivered at least once?

Reply:
>Does this mean that they're guaranteed to be delivered at least once?
Yes, that is correct. A webhook will be delivered at-least-once. And that means there is a high chance that notification will be delivered again as a duplicate.